### PR TITLE
feat: Make yamq dependency-free

### PR DESCRIPTION
**Pull request by moritzploss-k**
## About

Since more and more applications are moving away from `lager` (and possibly `stdlib2`), it's useful to have a dependency-free version of `gen_lb`. I'm also adding a GitHub actions CI flow, and suggest to switch from `erlang.mk` to `rebar3`.

### CODE REVIEW

File: src/gen_lb.erl [hunk 1]
```diff
@@ -258,30 +255,43 @@ cluster_maybe_block(Up0, Down, MaxErrors, MaxCrasches) ->
...
+  case lists:keyfind(K, 1, L) of
+    {K, V} -> {ok, V};
+    false  -> {error, notfound}
+  end.
+
+assoc(KVs, K, Def) ->
+  case assoc(KVs, K) of
+    {ok, V}           -> V;
+    {error, notfound} -> Def
+  end.
```
Comment (maxno-kivra) [inline src/gen_lb.erl]: I prefer self describing variable names, aka not shorten. The one I found most confusing of these is `Def`, first I read `Definition` and then `Default`. Please expand these, we can afford the extra bytes 😉

File: src/yamq.erl [hunk 2]
```diff
@@ -364,15 +365,21 @@ spawn_store(Store, Task, Options, From) ->
...
         ok   = Store:put(Store:encode_key(Info), Task),
         ok   = gen_server:cast(Daddy, {enqueued, {erlang:self(), Info}}),
         _    = gen_server:reply(From, ok) %tell caller we are done
     end).
 
+assoc(L, K) ->
+  case lists:keyfind(K, 1, L) of
+    {K, V} -> {ok, V};
+    false  -> {error, notfound}
+  end.
```
Comment (maxno-kivra) [inline src/yamq.erl]: Same as for the other `assoc`. I would love to see unambiguous variable names.

**Review by maxno-kivra — CHANGES_REQUESTED** [review]
I love that you're ripping out stdlib2. About time I say. 

I just have one small nitpick, then this is good to go.

(One could argue to change all `s2_lisits:assoc` to `lists:keyfind` but I get that you want to minimize the diff and that's a very good argument)

File: src/yamq.erl (hunk 2 above)
Comment (moritzploss-k) [reply src/yamq.erl]: sure, will change that! It was literally a copy-paste from stdlib2, but you're right, I should have given it some extra love

**Review by maxno-kivra — APPROVED** [review]
🚀

**Pull request merged**